test/hotspot/jtreg/compiler/intrinsics/float16/TestFP16ScalarAdd.java

```diff
@@ -57,6 +55,7 @@ public TestFP16ScalarAdd() {
 
     @Test
     @IR(applyIfCPUFeature = {"avx512_fp16", "true"}, counts = {IRNode.ADD_HF, "> 0", IRNode.REINTERPRET_S2HF, "> 0", IRNode.REINTERPRET_HF2S, "> 0"})
+    @IR(applyIfCPUFeatureAnd = {"fphp", "true", "asimdhp", "true"}, counts = {IRNode.ADD_HF, "> 0", IRNode.REINTERPRET_S2HF, "> 0", IRNode.REINTERPRET_HF2S, "> 0"})
```

Comment:
Please have one IR rules with unified feature list for x86 and aarach64. 
Not sure, but asimdhp is only needed for SIMD instructions I guess.

Reply:
Thanks @jatin-bhateja for your comments.
On Neon (non-SVE) machines or when we are testing scalar FP16 operations, Float16 is supported only if both `fphp` and `asimdhp` features are supported. So if we have to write it as a single rule, it needs to be something like - `avx512_fp16 == true || (fphp == true && asimdhp == true)` and I wasn't sure if this can be done in single IR rule here. Where ever I could write it as a single rule, I have done that for ex. clubbing avx_512_fp16 and sve in the same IR rule for `TestFloat16VectorSum.java`.

Reply:
I see, only way i can think of circumventing it currently is by creating a different test for AARCH16 as done for [some other tests](https://github.com/openjdk/jdk/tree/master/test/hotspot/jtreg/compiler/vectorapi/reshape)
 * @requires vm.cpu.features ~= ".*avx512bw.*"
 * @requires vm.cpu.features ~= ".*sve.*"

Reply:
Current rules looks ok to me, I just wrongly read the results on x86 where one of the IR rules failed, and that is bound to happene since these features are AARCH64 specific.